### ISSUE TITLE
[BUG] Import not working in Firefox (desktop)

### ISSUE THREAD

**Issue by Gitoffthelawn** [darkreader/darkreader#3927]
### **Describe the bug**

Import Not Working in Firefox (Desktop)

### **Steps to reproduce**

- Turn on the new version preview mode
- Export all settings
- Reset all settings
- Import all settings

### **Expected behavior**

Expect settings to be imported.

### **Actual behavior**

Settings not imported at all.

### **System Information:**
 - OS: Windows
 - Browser: Firefox 82
 - Dark Reader version: 4.9.21

### **Additional context**

I see #3273 for import in Firefox for Android not working, and #3899 for import in Opera not working, but this bug report is for import not working in the desktop version of Firefox.

Is there any workaround to currently import a valid Dark Reader settings file in Firefox?  As you can imagine, it's a big setback to lose all settings and not be able to import them.

**Comment by Gusted** [comment]
Hi @Gitoffthelawn!

Seems they have changed their UI popup for extensions behavior in Firefox V82 and need some new workaround.
Also it somehow has the same underlying issue as Opera https://github.com/darkreader/darkreader/issues/3899#issuecomment-713779246

Regards,
Gusted

**Comment by Gitoffthelawn** [comment]
Greetings Gusted!

I'm glad it was easy to confirm.  Outside of implementing a change within the extension itself, is there any way in Firefox 82+ to currently import a valid DR export file?

**Comment by Gusted** [comment]
Hi @Gitoffthelawn!

Their is! but a bit tricky.
So go to `about:debugging#/runtime/this-firefox`
Then scroll to the dark reader extension.
Click on `inspect manifest`.
Then in the new tab, replace IN the url `manifest.json` with `"ui/popup/index.html`.

Regards,
Gusted

**Comment by Gitoffthelawn** [comment]
Thank you so much!

The instructions may be a touch off (for me, the manifest didn't come up in the URL) , but close enough for me to understand what you are recommending... and, most importantly, it works great!

Here's another real simple way to accomplish the same task:

1.  Right-click on the DR popup
1.  Select `View Page Source` from the context menu
1.  Remove `view-source:` from the beginning of the URL in the address bar in the new source viewer tab

Voilà!

**Comment by Gitoffthelawn** [comment]
Just a follow-up: Issue remains in Dark Reader v4.9.26

**Comment by Gusted** [comment]
Hi @Gitoffthelawn!

Thanks for the follow-up! I've opened up a PR(#4724) to add a workaround for Firefox's behaviour and shouldn't be a extra hassle over chrome users.

Regards,
Gusted

**Comment by Gitoffthelawn** [comment]
Hi @Gusted!

You're welcome, and thank you for the PR!

**Comment by xcasxcursex** [comment]
Slightly updated instructions based on the above:

> Paste the following into the address bar and press enter: `about:debugging#/runtime/this-firefox`
> Scroll down to the dark reader extension. You will see the entry with four rows, one with a button and three with text.
> The bottom of the four rows is titled "Manifest URL" and on the right there is a URL which will look something like `moz-extension://00b374c7-bf4a-43b8-a80c-da0b8b8c84d4/manifest.json`. Click that link, it will open a new tab with a lot of text (the JSON file).
> In the new tab's address bar, you will see the URL described above. Replace `manifest.json` with `ui/popup/index.html`, using the above example, our URL would now be ``moz-extension://00b374c7-bf4a-43b8-a80c-da0b8b8c84d4/ui/popup/index.html`. Press enter. This will open a page which looks exactly like the normal Dark Reader settings popup.

You can now import the exported settings as normal.

**Comment by JAK0723** [comment]
I tried this workaround to import my chrome settings into firefox, but it doesn't seem to be working. The import does seem to succeed as the site list is updated in `ui/popup/index.html`, however if I modify any settings then everything gets reverted to before the import. Additionally, the extension in my toolbar is unchanged throughout this process.

**Comment by Miteirao** [comment]
Firefox Android I can't export settings

**Comment by xcasxcursex** [comment]
I exported my settings tonight, reset settings, and the entire UI has changed into one which lacks the option to revert to the old one, or import my saved settings.... That is to say, it's now not just 'not working', it's 'gone'.

Lil help? I'd kinda really like to get those settings back.

**Comment by Gusted** [comment]
Because of our current message handling, the workaround I had made a while back doesn't work anymore. So I've opened a bug report over at mozilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1762774

**Comment by Gitoffthelawn** [comment]
It looks like they marked it as a duplicate of this bug that was reported 5 years ago by another dedicated extension developer: https://bugzilla.mozilla.org/show_bug.cgi?id=1378527

Looking at that other bug report, the Mozilla development team hasn't even changed the bug's status from **"UNCONFIRMED"**, nor assigned a developer to fix this well-documented Firefox bug.

It's a bit frustrating how many developer hours they dedicated to "theme refreshes" that nobody wanted instead of dedicating sufficient resources to even confirm (never mind *fix*) reported bugs like this one. :(  Mozilla continues to suffer from poor management that doesn't seem capable of setting obvious priorities.  A quick glance at their salaries reveals that they are paid extravagant sums of money to continue to make bad decisions.  Personally, I stopped donating to Mozilla because of this sort of thing...

**Comment by Gusted** [comment]
Aside from Mozilla's management. It's indeed frustrating that this is still open. But so far I saw with Firefox bugs, if some big site/extension also suffers from the bug it magically get some push to work on it.

I will/have added a comment on the bug report for it.

**Comment by Gitoffthelawn** [comment]
Thanks.  I think your comment on Bugzilla is very useful.  I'm not sure if it will help, but you might want to also make it abundantly clear that the bug affects Dark Reader, which is a Mozilla Recommended extension used by hundreds of thousands of people.  Also, this Firefox bug affects at least one extension by sblask who also creates high-quality extensions.

**Comment by Gusted** [comment]
> Thanks. I think your comment on Bugzilla is very useful. I'm not sure if it will help, but you might want to also make it abundantly clear that the bug affects Dark Reader, which is a Mozilla Recommended extension used by hundreds of thousands of people. Also, this Firefox bug affects at least one extension by sblask who also creates high-quality extensions.

Yeah, from what I see of the comments is that triager didn't have a clue where the bug should belong(and currently seems to live in limbo). Maybe a new refreshing look a few years later can help giving the right push to the correct responsible code owner, giving a issue some time does wonderful things!

**Comment by gluons** [comment]
For me, I use this workaround.

Open `moz-extension://<add-on id>/ui/popup/index.html` in new tab then import it.  
(You can find **`add-on id`** by open `about:debugging#/runtime/this-firefox` in Firefox.)

![ภาพ](https://user-images.githubusercontent.com/4688092/177102320-08311f7e-db2d-4dce-8577-995d8cffaa28.png)

**Comment by Miteirao** [comment]
> For me, I use this workaround.
> 
> Open `moz-extension://<add-on id>/ui/popup/index.html` in new tab then import it. (You can find **`add-on id`** by open `about:debugging#/runtime/this-firefox` in Firefox.)
> 
> ![ภาพ](https://user-images.githubusercontent.com/4688092/177102320-08311f7e-db2d-4dce-8577-995d8cffaa28.png)

Yes, it worked for export settings on Firefox for Android! I found an easier way:
Go to settings, then click on "Make a fix"
![Screenshot_2022-07-25-05-16-21-51_3aea4af51f236e4932235fdada7d1643](https://user-images.githubusercontent.com/88121690/180731465-0c3fd3de-2140-4d8b-aad1-d4dcf56d5548.jpg)
It will open in a new tab with URL bar available, so replace `devtools` with `popup` and you can now export settings file to Download folder.

**Comment by jluo98** [comment]
Disabling Popup Auto-Hide also does the trick. 

![image](https://user-images.githubusercontent.com/8736917/200492779-593de863-e1b0-4d3d-9853-7c4e2611d558.png)

**Comment by lapo-luchini** [comment]
> Looking at that other bug report, the Mozilla development team hasn't even changed the bug's status from **"UNCONFIRMED"**, nor assigned a developer to fix this well-documented Firefox bug.

Was [finally confirmed on 2023-05-17](https://bugzilla.mozilla.org/show_bug.cgi?id=1378527#c25), with two attached test web-extensions.
Then in stopped at that, again…
